File: cpp/1768_merge_strings_alternately.cpp
/*
https://leetcode.com/problems/merge-strings-alternately/submissions/1108567638/
*/

class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string result = "";
        int i = 0;
        int j = 0;
        int n = word1.size();
        int m = word2.size();
        while (i < n || j < m) {
            if (i < n) {
                result.push_back(word1[i]);
                i++;
            }
            if (j < m) {
                result.push_back(word2[j]);
                j++;
            }
        }
        return result;
    }
};

File: rust/1768_merge_strings_alternately.rs
'''
https://leetcode.com/problems/merge-strings-alternately/submissions/1109084194/?envType=study-plan-v2&envId=leetcode-75
'''

impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut result = String::with_capacity(word1.len() + word2.len());
        
        let mut chars1 = word1.chars();
        let mut chars2 = word2.chars();

        for i in 0..word1.len().max(word2.len()) {
            match chars1.next() {
                Some(char) => { result.push(char); },
                _ => {}
            }
            match chars2.next() {
                Some(char) => { result.push(char); },
                _ => {}
            }
        }
        result
    }
}